src/service/identity_service.rs

```diff
@@ -196,10 +186,10 @@ impl IdentityServiceApi for IdentityService {
             postal_address,
             public_key_pem,
             private_key_pem,
-            bitcoin_public_key: public_key,
-            bitcoin_private_key: private_key.clone(),
-            nostr_npub: None,
-            nostr_relay: None,
+            bitcoin_public_key: public_key.to_string(),
```

Comment:
Yes that is probably a good idea, also with the separate task. It might be some rewrite but would prevent errors I already encountered (on certain failures) where the Bitcoin key was invalid bcs of 0 length (empty string). So identity access without it being created before.